### PR TITLE
[bugfix] Make enable_torch_compile_vae actually compile the Wan VAE

### PR DESCRIPTION
**Pull request by KyleNeverGivesUp**
## Purpose

`enable_torch_compile_vae=True` is silently a no-op for the Wan VAE (`AutoencoderKLWan`): the compiled module is built but never executed during decode.

## Changes

**Root cause:** `DecodingStage` is constructed with a direct reference to the VAE module, then `_maybe_compile_pipeline_module` replaces `self.modules["vae"]` with a `torch.compile`-wrapped object. Because `AutoencoderKLWan` had no `_compile_conditions`, it took the full-module-replace fallback, so the stage kept using the original uncompiled reference and the compiled VAE never ran. (`LTX2CausalVideoAutoencoder` avoids this via `_compile_conditions`, which compiles the codec `forward` in place.)

- Add `_compile_conditions` to `AutoencoderKLWan` matching its `encoder`/`decoder` submodules, so compilation happens in place and the existing `DecodingStage` reference uses it.
- Add a lightweight (no-GPU) regression test that the predicate is registered and matches the codec submodules.

## Test Plan

```bash
# Unit test (no GPU / no weights)
pytest fastvideo/tests/vaes/test_wan_vae_compile.py -v

# End-to-end profiling: FastWan2.1-T2V-1.3B, 3-step DMD, GB10 (sm_121),
# enable_torch_compile=True; enable_torch_compile_vae=True before vs after this fix.
nsys profile -t cuda,nvtx -o run python run_infer.py
nsys stats --report cuda_gpu_kern_sum run.nsys-rep
```

## Test Results

Unit test: 3 passed (registers `_compile_conditions` / matches encoder-decoder / rejects others).

GENERATE-phase GPU kernels (DiT `torch.compile` on in both rows; only the VAE differs):

| Config | generate (s) | elementwise (ms) | total kernels |
|---|---|---|---|
| flag on, no `_compile_conditions` (no-op) | 26.3 | 13107 | 17598 |
| flag on, **this PR** | 24.1 | 10027 | 15854 |

- Before the fix the kernel count is identical to not setting the flag at all (the no-op). After, the log shows the in-place path: `Enabled torch.compile for 1 submodules in vae via _compile_conditions`.
- VAE-decode elementwise fuses (−3080 ms, −1744 kernels), ~8% faster end-to-end on top of DiT compile.
- **Quality:** MS-SSIM **0.9910** between the uncompiled-VAE and compiled-VAE outputs at identical seed/DiT (isolates the VAE change) — near-lossless. Measured with the repo's `compute_video_ssim_torchvision`.

## Checklist

- [x] I ran `pre-commit run` (changed source is under `fastvideo/models/`, which is excluded from pre-commit by design)
- [x] I added or updated tests for my changes
- [x] I considered GPU memory impact of my changes

### CODE REVIEW

File: fastvideo/models/vaes/wanvae.py [hunk 1]
```diff
@@ -1087,13 +1087,27 @@ def unpatchify(x, patch_size):
...
+def _is_wan_vae_codec(name: str, submodule: nn.Module) -> bool:
+    """Match the Wan VAE's encoder/decoder for in-place torch.compile.
+
+    Compiling the codec submodules' ``forward`` in place (rather than replacing
+    the whole VAE module) ensures the compiled version is the one held by the
+    already-constructed DecodingStage. Without this, ``enable_torch_compile_vae``
+    falls back to replacing ``self.modules["vae"]`` and is a no-op for Wan.
+    """
+    return name in {"encoder", "decoder"} and isinstance(
+        submodule, (WanEncoder3d, WanDecoder3d))
```
[the remark this answers is not in the document]
Comment (KyleNeverGivesUp) [reply fastvideo/models/vaes/wanvae.py]: Done, switched to a strict name, type mapping and added mismatch rejection tests.

File: fastvideo/tests/vaes/test_wan_vae_compile.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+def test_is_wan_vae_codec_rejects_other_submodules():
+    decoder = object.__new__(WanDecoder3d)
+
+    # Right name but wrong type (e.g. a small conv/linear at the same name).
+    assert _is_wan_vae_codec("decoder", nn.Linear(1, 1)) is False
+    # Right type but not the top-level codec name.
+    assert _is_wan_vae_codec("decoder.something", decoder) is False
+    assert _is_wan_vae_codec("post_quant_conv", decoder) is False
```
[the remark this answers is not in the document]
Comment (KyleNeverGivesUp) [reply fastvideo/tests/vaes/test_wan_vae_compile.py:45]: Done

**Comment by Satyam-53** [comment]
@KyleNeverGivesUp Good one finding this bug and raising the PR, I did not get time to inspect code but I wasn't getting any performance boost with vae_compile on the existing code. Hope this fix would make me see the boost.

**Comment by Satyam-53** [comment]
Thanks, this approach looks right to me. I verified the core behavior locally with the `fastvideo` conda env: `torch.compile(vae)` compiles the module `forward` / `__call__` path, but calling `vae.decode(...)` remains eager. Since `DecodingStage` calls `self.vae.decode(latents)`, the old fallback compiled `AutoencoderKLWan.forward` but did not affect the inference decode path.

The `_compile_conditions` fix is the right shape because it compiles the heavy codec submodules in place:

- `encoder.forward`
- `decoder.forward`

Then the normal path:

```python
vae.decode(latents) -> self.decoder(...) -> compiled decoder.forward(...)
```
actually uses compiled code.

One small thing I’d like adjusted before merge: the new explanatory comment/docstring says the issue is that DecodingStage held the original VAE reference. In this PR base, pipeline compile runs before stages are created, so the more precise issue is that full-module torch.compile(vae) only compiles forward, while inference calls decode.

Could you update that wording to say something like:
```
DecodingStage calls AutoencoderKLWan.decode(), while the fallback torch.compile(vae) only compiles forward; compiling the encoder/decoder submodule forwards in place makes the encode/decode paths hit compiled code.
```

**Review by SolitaryThinker — APPROVED** [review]
